src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/StaffCategoryCaseTest.java

```diff
@@ -494,6 +496,167 @@ void shouldDeleteApprovedRequestedRoleForCase_V1_1() {
+        assignmentRequest.setRequestedRoles(List.of(requestedRole1));
+        assignmentRequest.getRequest().setClientId("ccd_data");
+        assignmentRequest.getRequest().setByPassOrgDroolRule(enableByPass);
+
+        FeatureFlag featureFlag  =  FeatureFlag.builder().flagName(FeatureFlagEnum.CCD_1_0.getValue())
+            .status(true).build();
+        featureFlags.add(featureFlag);
+
+        executeDroolRules(List.of(buildExistingRoleForIAC(requestedRole1.getActorId(),
+                                                          "senior-tribunal-caseworker"),
+                                  buildExistingRoleForIAC(assignmentRequest.getRequest().getAssignerId(),
+                                                          "tribunal-caseworker"),
+                                  buildExistingRoleForIAC(assignmentRequest.getRequest().getAssignerId(),
+                                                          "[RESPSOLICITOR]")));
+        //assertion
+        assignmentRequest.getRequestedRoles().forEach(ra -> assertEquals(status, ra.getStatus()));
+    }
+
+    @Test
+    void shouldApproveDeleteCaseRequestedRoles_CCD_1_0_FromCCD() {
+        verifyDeleteCaseRequestRole_CCD_1_0("[PETSOLICITOR]", "ccd_data", RoleCategory.PROFESSIONAL);
+        verifyDeleteCaseRequestRole_CCD_1_0("[CREATOR]", "aac_manage_case_assignment",
+                                            RoleCategory.CITIZEN);
+    }
+
+    @Test
+    void shouldApproveDeleteCreaterCaseRequestedRoles_CCD_1_0_FromCCD() {
+        verifyDeleteCaseRequestRole_CCD_1_0("[CREATOR]", "ccd_data", RoleCategory.JUDICIAL);
+        verifyDeleteCaseRequestRole_CCD_1_0("[CREATOR]", "aac_manage_case_assignment",
+                                            RoleCategory.LEGAL_OPERATIONS);
+    }
+
+    private void verifyDeleteCaseRequestRole_CCD_1_0(String roleName, String clientId, RoleCategory category) {
+        RoleAssignment requestedRole1 = getRequestedCaseRole(category, roleName,
+                                                             SPECIFIC, "caseId",
+                                                             "1234567890123456", DELETE_REQUESTED);
+        requestedRole1.setClassification(Classification.RESTRICTED);
+        requestedRole1.getAttributes().putAll(Map.of("jurisdiction", convertValueJsonNode("IA"),
+                                                     "caseType", convertValueJsonNode("Asylum"),
+                                                     "caseId", convertValueJsonNode("1234567890123456")));
+        assignmentRequest.setRequestedRoles(List.of(requestedRole1));
+        assignmentRequest.getRequest().setClientId(clientId);
+
+        FeatureFlag featureFlag  =  FeatureFlag.builder().flagName(FeatureFlagEnum.CCD_1_0.getValue())
+            .status(true).build();
+        featureFlags.add(featureFlag);
+
+        executeDroolRules(List.of(buildExistingRoleForIAC(requestedRole1.getActorId(),
```

Comment:
Use buildExecuteKieSession() instead of executeDroolRule(List< ExistingRoleAssignments>) as we don't need to inject ExistingRoleAssignments to validate CCD Roles in drool. it will unnecessary consume the system memory.